src/irmin-pack/inode.ml

```diff
@@ -51,10 +50,12 @@ end
 
 module Make
     (Conf : CONFIG)
-    (Pack : Pack.MAKER)
-    (H : Irmin.Hash.S with type t = Pack.key)
+    (H : Irmin.Hash.S)
+    (Pack : Pack.MAKER with type key = H.t)
```

Comment:
I'm not sure you need to change the order of parameters anymore

Reply:
I think we do still need to change it for the same reason as we need the index constraint [here](https://github.com/mirage/irmin/pull/805/files#r305413487).